protected/controllers/PatientController.php

```diff
@@ -1585,9 +1594,26 @@ private function performPatientSave(Contact $contact, Patient $patient, Address
 
                         if ($referral->save()) {
 
-                            $transaction->commit();
+                            if (isset($patientuserreferral) && $patientuserreferral->user_id !='')
```

Comment:
As discussed before, this should ideally be moved into its own function so the same logic doesn't need to be duplicated.

As this currently works, there is no need to do this immediately but it would be ideal to implement a function in a future code-cleaning exercise.